portfolio/src/main/webapp/script.js

```diff
@@ -28,29 +28,41 @@ function fetchFunction() {
     })
 }
 
-function createNewElement({emoji, name, text, id}) {
+function createNewElement({key, emoji, name, text, id}) {
     const newElement = document.createElement('div');
     newElement.setAttribute('class', 'media margin10px');
     newElement.setAttribute('id', id);
     newElement.innerHTML = `
         <img class='margin10px' ${emoji == 1 ? ("src='icons/emoji-smile.svg'") : 
             (emoji == 2 ? "src='icons/emoji-neutral.svg'" : "src='icons/emoji-frown.svg'")} 
-            width="32" height="32" color/>
+            width="32" height="32"/>
         <div class='media-body'>
             <h5 class='mt-0'>${name}</h5>
             ${text}
         </div>
+        <button class="btn" onClick="deleteCommentByKey('${key}')">
+            <img class='autoMargin' src='icons/trash.svg' width="20" height="20"/>
+        </button>
     `;
     return newElement;
 }
 
 function deleteComments() {
     const request = new Request('/delete-data', {method: 'POST'});
+    fetchRequestAndReload(request);
+}
+
+function deleteCommentByKey(key) {
+    const request = new Request(`/delete-comment?key=${key}`, {method: 'POST'} );
```

Comment:
Can the key be moved into the request body instead?

Reply:
Yes, I guess it could be moved, I don't know how to get it from the java servlet if its in the body but let me look into it.